interlay/interbtc-ui
# feat: add parity signer companion for polkadot vault support

# Interbtc UI Pull Request Template

## Description

- Signing raw data as we require for the T&Cs is not supported in the polkadot.js extension version released in the Firefox and Chrome stores: https://github.com/polkadot-js/extension/issues/1268
- Users that use the Polkadot Vault hardware wallet (https://signer.parity.io/) cannot sign the T&Cs and thus not use our app. Once they signed the T&Cs, these users can use polkadot.js in combination with the Polkadot Vault to sign transactions
- This PR adds https://github.com/paritytech/parity-signer-companion as a wallet option that allows signing raw data
- Users can use the parity signer companion extension to sign T&Cs and then choose to continue using the parity signer companion or the polkadot.js extension

## Current behaviour (updates)

The possible wallets to select are polkadot.js, talisman, and subwallet.

## New behaviour

Add the parity signer companion.

![image](https://github.com/interlay/interbtc-ui/assets/14966470/24117963-1b73-4a4f-ba0e-273e13aaeb3f)

## Reproducible testing steps:

1. Open the select wallet modal and see Parity Signer Companion as an option
2. Install Parity Signer Companion and add a wallet. See that you can sign the T&Cs
